odoo/addons/base/wizard/base_partner_merge.py

```diff
@@ -234,6 +234,32 @@ def update_records(model, src, field_model='model', field_id='res_id'):
                 dest_record_id=dst_record.id,
             ))
 
+            # company dependent fields referring the merged records might not be set on the table,
+            # we need to update the default values
+            self.env.cr.execute(SQL(
+                """
+                UPDATE ir_default
+                SET json_value =
+                    CASE
+                        WHEN json_value::int IN %(src_record_ids)s
+                        THEN %(dest_record_id)s::varchar
+                        ELSE json_value::varchar
```

Comment:
> We should provide the correct format for SET for readability, json_value is a char column, please use text format even if postgresql can auto convert it.

Something like that?

Reply:
I propose to use
```suggestion
                        THEN %(dest_record_id_str)s
                        ELSE json_value
```
and we can use
`dest_record_id_str=str(dst_record.id),`